src/dx/debug_menu.c

```diff
@@ -345,4 +347,94 @@ void dx_debug_menu_cb_decreasehammer(void* args){
     }
 }
 
+u8 cb_currentpartner = 0;
+
+void dx_debug_menu_cb_partnerincreaselevel(void* args){
+    if(gPlayerData.partners[*(u8*)args].level < 2 ) {
+        gPlayerData.partners[*(u8*)args].level++;
+    }
+}
+void dx_debug_menu_cb_partnerdecreaselevel(void* args){
+    if(gPlayerData.partners[*(u8*)args].level > 0 ) {
+        gPlayerData.partners[*(u8*)args].level--;
+    }
+}
+
+void dx_debug_menu_cb_partnerenable(void* args){
+    if(gPlayerData.partners[*(u8*)args].enabled != 1 ) {
+        gPlayerData.partners[*(u8*)args].enabled = 1;
+    }
+}
+
+void dx_debug_menu_cb_partnerdisable(void* args){
+    if(gPlayerData.partners[*(u8*)args].enabled != 0 ) {
+        gPlayerData.partners[*(u8*)args].enabled = 0;
+    }
+}
+
+void dx_debug_menu_cb_partnermenu(void* args){
+    struct dx_debug_menu_item* items;
+        dx_debug_menu_push();
+         cb_currentpartner = *(u8*)args; //current partner in the menu
+    dx_debugMenu.title = "partner options";
+    items = general_heap_malloc(5 * sizeof(struct dx_debug_menu_item));
+    dx_debugMenu.items = items;
+    items[0].name = "Increase Level";
+    items[0].callback = dx_debug_menu_cb_partnerincreaselevel;
+    items[0].callbackArg = &cb_currentpartner;
+    items[1].name = "Decrease Level";
+    items[1].callback = dx_debug_menu_cb_partnerdecreaselevel;
+    items[1].callbackArg = &cb_currentpartner;
+    items[2].name = "Enable Partner";
+    items[2].callback = dx_debug_menu_cb_partnerenable;
+    items[2].callbackArg = &cb_currentpartner;
+    items[3].name = "Disable Partner";
+    items[3].callback = dx_debug_menu_cb_partnerdisable;
+    items[3].callbackArg = &cb_currentpartner;
+    items[4].name = NULL;
+}
+u8 selectedpartner[] = {PARTNER_GOOMBARIO, PARTNER_KOOPER, PARTNER_BOMBETTE, PARTNER_PARAKARRY, PARTNER_GOOMPA, PARTNER_WATT, PARTNER_SUSHIE, PARTNER_LAKILESTER, PARTNER_BOW, PARTNER_GOOMBARIA, PARTNER_TWINK}; // partner enums
```

Comment:
Interesting... did you do this because `callbackArg` had to be a `void*`? If so, you could just do

```diff
-items[0].callbackArg = &selectedpartner[0];
+items[0].callbackArg = (void*)PARTNER_GOOMBARIO;
```

and cast, not dereference, `arg` in the callback:

```diff
 void dx_debug_menu_cb_partnerincreaselevel(void* args){
+     // args should really be called 'arg'...
-     if(gPlayerData.partners[*(u8*)args].level < 2 ) {
-        gPlayerData.partners[*(u8*)args].level++;
+     if(gPlayerData.partners[(u8)args].level < 2 ) {
+        gPlayerData.partners[(u8)args].level++;
     }
 }
```